app/models/like.rb

```diff
@@ -0,0 +1,9 @@
+class Like < ApplicationRecord
+  belongs_to :post
+  belongs_to :author
+  validates :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+
+  def increment_like_count
+    post.increment!(:likes_counter)
+  end
```

Comment:
- [ Required ] I think it would be better if you use [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#increment_like_count` method every time a new like has been saved to database, then make the `increment_like_count` method private. Please see the example below and rectify this.

## Example code 
```ruby
  after_save :increment_like_count

  private

  def increment_like_count
    post.increment!(:likes_counter)
  end
```